### PR TITLE
Dueling mechanic

### PR DESCRIPTION
**Pull request by EricR401S**
I created the mechanic, and the behavior I expect is there.

I have not passed the mypy checks and so forth. I wonder, @rjrichards27 , did your original file pass mypy's dungeon?

### CODE REVIEW

**Comment by rjrichards27** [comment]
Yes, my original file passed all mypy tests.

File: src/game.py [hunk 1]
```diff
@@ -23,6 +23,78 @@ def print_duel_pairs(self, character_pairs_list: list[dict[str, str]]) -> None:
             print("Duel: ", duel_pairs[0]["Name"], "vs.", duel_pairs[1]["Name"])
             print("   ")
 
+    def duel(
+        self,
+        character_group: tuple[dict[str, str], dict[str, str]],
+        spells: list[dict[str, str]],
+    ) -> int:
+        """Simulates a duel between two characters.
```
Comment (rjrichards27) [inline src/game.py:31]: Very detailed docstring!

File: src/game.py [hunk 2]
```diff
@@ -23,6 +23,78 @@ def print_duel_pairs(self, character_pairs_list: list[dict[str, str]]) -> None:
...
+                + int(character["Emotional Strength"])
+            )
+
+            # the random choice and the popping of that choice
+            # will impede duplicate spells from being used
+            spell_num = random.randint(0, len(spells_copy))
+            if spell_num == len(spells_copy):
+                # this means the character used no spells
+                spell_stats = 0
+                chosen_spell = {"Name": "no spell", "Power": "0", "Description": ""}
```
Comment (rjrichards27) [inline src/game.py:65]: Good idea doing this. I would not have thought of that.

File: src/game.py [hunk 3]
```diff
@@ -23,6 +23,78 @@ def print_duel_pairs(self, character_pairs_list: list[dict[str, str]]) -> None:
...
+
+            else:
+                # this means the character used a spell
+                chosen_spell = spells_copy.pop(spell_num)
+                spell_stats = int(chosen_spell["Power"])
+
+            battle_stats[i] = character_stats + spell_stats
+            spells_used[i] = chosen_spell
+
+            print(
```
Comment (rjrichards27) [inline src/game.py:76]: Does this only print for the character that wins or both characters?

File: src/test_duel.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+
+def test_duel() -> None:
+    """Test duel mechanic.
+
+    This method tests the duel mechanic of the game.
+    It is designed to work with the pairs of characters
+    that are fated to battle each other."""
+
+    assert type(game1.duel(battle_pair[0], test_spells)) is int
```
Comment (rjrichards27) [inline src/test_duel.py:34]: Should this be more specific?

**Review by rjrichards27 — COMMENTED** [review]
Added my comments

**Comment by wafiakmal** [comment]
I have no comment on the code. Should I merge this PR, or are there any changes that @EricR401S wants to incorporate based on Rachel's comments?

**Pull request merged**